### PR TITLE
Fix for IF function in the order by clause and handle hexadecimal literals.

### PR DESCRIPTION
**Pull request by tntnatbry**
Fixes working_tpch1/misc/CCBillEU.sql and working_tpch1_compareLogOnly/view/mts_view.40.sql.

### CODE REVIEW

File: dbcon/mysql/ha_calpont_execplan.cpp [hunk 1]
```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
 
                 case STRING_RESULT:
                 {
+                    // Special handling for 0xHHHH literals
+                    const Type_handler *tph = item->type_handler();
+                    if (dynamic_cast<const Type_handler_hex_hybrid*>(tph) != NULL)
```
Comment (drrtuy) [inline dbcon/mysql/ha_calpont_execplan.cpp]: That is cheaper to use typeid(*tph)==typeid(Type_handler_hex_hybrid) b/c dynamic_cast recurses through the class hierarchy. However tph could be of a different class that inherits from Type_handler_hex_hybrid. If the suggestion is applicable then use it.

File: dbcon/mysql/ha_calpont_execplan.cpp [hunk 2]
```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
...
+                    {
+                        Item_hex_hybrid *hip = (Item_hex_hybrid*) item;
+                        gwip->rcWorkStack.push(new ConstantColumn((int64_t)hip->val_int(), ConstantColumn::NUM));
+                        (dynamic_cast<ConstantColumn*>(gwip->rcWorkStack.top()))->timeZone(gwip->thd->variables.time_zone->get_name()->ptr());
```
Comment (drrtuy) [inline dbcon/mysql/ha_calpont_execplan.cpp]: Heavy to read. Kindly separate operations to make it readable.

File: dbcon/mysql/ha_calpont_execplan.cpp [hunk 3]
```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
 
                 case STRING_RESULT:
                 {
+                    // Special handling for 0xHHHH literals
+                    const Type_handler *tph = item->type_handler();
+                    if (dynamic_cast<const Type_handler_hex_hybrid*>(tph) != NULL)
+                    {
+                        Item_hex_hybrid *hip = (Item_hex_hybrid*) item;
```
Comment (drrtuy) [inline dbcon/mysql/ha_calpont_execplan.cpp]: I know that MDB code uses c-type cast but we should stick with cpp-style reinterpret_cast. Feel free to tell me if I use c-style cast accidently.

File: dbcon/mysql/ha_calpont_execplan.cpp (hunk 1 above)
Comment (tntnatbry) [reply dbcon/mysql/ha_calpont_execplan.cpp]: This is a good idea. Done.

File: dbcon/mysql/ha_calpont_execplan.cpp (hunk 2 above)
Comment (tntnatbry) [reply dbcon/mysql/ha_calpont_execplan.cpp]: Done.

File: dbcon/mysql/ha_calpont_execplan.cpp (hunk 3 above)
Comment (tntnatbry) [reply dbcon/mysql/ha_calpont_execplan.cpp]: Done. I could not directly use reinterpret_cast as the compiler complained about the constness of item (as it is passed as const Item* to gp_walk()). So I had to first cast away the constness using const_cast.

**Comment by tntnatbry** [comment]
PR is updated to incorporate the feedback. Please review.

**Review by drrtuy — APPROVED** [review]

**Pull request merged**